lib/instrumentation/index.js

```diff
@@ -234,7 +245,14 @@ Instrumentation.prototype._startHook = function () {
 
 Instrumentation.prototype._patchModule = function (exports, name, version, enabled) {
   this._agent.logger.debug('shimming %s@%s module', name, version)
-  var patches = this._patches.get(name)
+  const lambdaHandlerInfo = getLambdaHandler(process.env)
+  let patches
+  if (!lambdaHandlerInfo) {
+    patches = this._patches.get(name)
+  } else if (name === lambdaHandlerInfo.module) {
+    patches = this._patches.get(lambdaHandlerInfo.filePath)
+  }
```

Comment:
Doesn't this break all patching of other modules in a lambda env? `patches` will be undefined for them.

Reply:
Right you are -- looks like this inadvertently just checks for the existence of the handler instead of checking for both the existence of the handler AND whether the instrumentation currently being handled is the lambda instrumentation.  Test added to make sure the existing instrumentation are applied when lambda's detected and the correct (one hopes) logic's been added here: https://github.com/elastic/apm-agent-nodejs/pull/2577/commits/2e38f10fc9c4c4e6777af5fe6a74129cd411cd78#diff-413f1f1acaa8b1d97f8cae0884c443a777ef6eb822092e6483b4fe59a4936221R108